packages/editor/src/browser/editor-manager.ts

```diff
@@ -69,7 +69,13 @@ export class EditorManager extends WidgetOpenHandler<EditorWidget> {
     }
     protected updateActiveEditor(): void {
         const widget = this.shell.activeWidget;
-        this.setActiveEditor(widget instanceof EditorWidget ? widget : undefined);
+        if (widget instanceof EditorWidget) {
```

Comment:
@vinokurig why is it needed to extend editor manager that way? haven't seen that change.

would it have other implications? 
/cc @akosyakov

Reply:
Preferences editors are nested in the SplitPanel. If Focus changed `widget` will be SplitPanel so need to get editor from the SplitPanel

Reply:
Is it only because of the focus? Have you tried to set focus (or to activate) on the code editors when the host widget (SplitPanel) is activated?

Reply:
Yes I have, When performing any action from one of the preference editor `current` editor is retrieved but `current = undefined`  in that case

Reply:
Ah! Is that here because of keybindings?

Reply:
I'll try to look in this. Maybe we find a more generic way. I think it might be interesting for other parts to embed a code editor, without extending core, right?

Reply:
See my other comment. I think introducing a provider is fine in core.

Reply:
done